frontend/catalyst/jax_extras/tracing.py

```diff
@@ -536,6 +540,7 @@ def make_jaxpr_f(*args, **kwargs):
             in_type, in_tree = abstractify(args, kwargs)
             f, out_tree_promise = flatten_fun(f, in_tree)
             f = annotate(f, in_type)
+            f.f.static_argnums = static_argnums
```

Comment:
I don't like adding this attribute dynamically here. But I do think that the attribute needs to be owned by this object. :thinking: 

Do you know of another way to achieve this?

Reply:
Actually, I think your previous proposed method works fine! I changed it now. I was making a python mistake and thought it is not possible :)